functions/Copy-SqlIntegrationServices.ps1

```diff
@@ -0,0 +1,392 @@
+                $cmd.Parameters.Add("@folder_name",$Folder) | out-null;  
+                $cmd.Parameters.Add("@project_name",$Project) | out-null;  
+                [byte[]]$results = $cmd.ExecuteScalar();  
+                if($results -ne $null) {  
+                    $destFolder = $destinationFolders | ? { $_.Name -eq $Folder }
+                    $deployedProject = $destFolder.DeployProject($Project,$results)  
+                    Write-Output "Project: $Project - DeployStatus: $($deployedProject.Status)."
+                }  
+                else {  
+                    Write-Error "Failed deploying $Project from folder $Folder."
+                    continue  
+                } 
+            }
+            catch {
+                Write-Exception $_
+            } 
+        }
+
+        Function Create-Folder {
+            param(
+                [String]$Folder,
+                [String]$Description,
+                [Switch]$Force
+            )
+            if ($Force) {
+                $remove = $destinationFolders | ? { $_.Name -eq $Folder }
+                $envs = $remove.Environments.Name
+                foreach ($e in $envs) {
+                    $remove.Environments[$e].Drop()
+                }
+                $projs = $remove.Projects.Name
+                foreach ($p in $projs) {
+                    $remove.Projects[$p].Drop()
+                }
+                $remove.Drop()
+                $destinationCatalog.Alter()
+                $destinationCatalog.Refresh()
+            }
+            $destFolder = New-Object "$ISNamespace.CatalogFolder" ($destinationCatalog, $Folder, $Description)
+            $destFolder.Create()
+            $destFolder.Alter()
+            $destFolder.Refresh()
+        }
+        
+        Function Create-Environment {
+            param(
+                [String]$Folder,
+                [String]$Environment,
+                [Switch]$Force
+            )
+            $f = $destinationFolders | ? { $_.Name -eq $Folder }
+            if ($force) {
+                $f.Environments[$Environment].Drop()
+                $f.Alter()
+                $f.Refresh()
+            }
+            $srcEnv = ($sourceFolders | ? { $_.Name -eq $Folder }).Environments[$Environment]
+            $targetEnv = New-Object "$ISNamespace.EnvironmentInfo" ($f, $($srcEnv.Name), $($srcEnv.Description))
+            foreach ($var in $srcEnv.Variables) {
+                if ($var.Value.ToString() -eq "") { 
+                    $finalValue= ""
+                }
+                else {
+                    $finalValue = $var.Value
+                }
+                $targetEnv.Variables.Add($var.Name, $var.Type, $finalValue, $var.Sensitive, $var.Description)
+            }
+            $targetEnv.Create()
+            $targetEnv.Alter()
+            $targetEnv.Refresh()
+        }
+    }
+    PROCESS
+    {
+        if (!$sourceCatalog) {
+            Write-Error "The source SSISDB catalog does not exist."
```

Comment:
not sure if it's worth it.  as far as I know, SSISDB is the only available catalog name